articles/express/index.md

```diff
@@ -0,0 +1,373 @@
+app.use(express.static("static"); //relative path
+```
+
+Any requested files in the directory "static" are served. `localhost:3000/dummy_file.txt` will show the same result as above. We can call `static()` multiple times to use multiple static asset directories. For example, consider we have two directories `static` and `public` with static files and we wrote the following code:
+```JavaScript
+app.use(express.static("static");
+app.use(express.static("public");
+```
+Suppose you make a request like `localhost:3000/hello.html`, Express looks up the files in the `static` directory then `public` directory if the file exists then returns `hello.html`.
+
+
+### Routing
+Express makes request handling easier by mapping requests to different request handlers. A request handler is a function which handles all the requests to a specific path with a specific HTTP method. 
+
+In the basic example above, we saw how to handle a `GET` request. As an application grows in size the routes grow as well as do the request handlers.
+
+Lets see how we can use [Routers](http://expressjs.com/en/4x/api.html#router) to split a large app into smaller, maintainable functions. 
+
+According to the documentation, a Router is "an isolated instance of middleware and routes. Routers can be thought of as “mini” applications only capable of performing middleware and routing".
+
+Routers can be used like middleware functions, they can be added to middleware stack using the `app.use()` function. A simple example:
+
+```JavaScript
+//app.js the main file
+
+var express = require("express");
+var apiRouter = require("./routes/api_router");
+
+var app = express();
+
+app.use("/api", apiRouter);
+
+app.listen(3000);
+```
+Create a folder called `routes` and a file called `api_router.js` inside it.
+
+```JavaScript
+//routes/api_router.js
+var express = require("express");
+var router = express.Router();
+
+router.get("/route1", function(req, res, next){
+        res.send("Success !!");
+        next();
+});
+
+module.exports = router;
+
+```
+When you start the app and visit the URL `localhost:3000/api/route1` you can see `Success!!`. Take a look at all the router functions [here](http://expressjs.com/en/4x/api.html#router).
+
+#### Useful Routing Tips
+
+**Grabbing route parameters**
+
+Suppose you are building a website for a company that showcases their products, each product has a *productID*. You want the URL for product 1 to be `/product/1`. 
+
+Instead of defining a route for every product, you can define a single route for everything in the form of `product/productID` and then return a file based on the productID.
+
+```JavaScript
+var express = require("express");
+var app = express();
+
+//use a colon to grab a parameter
+
+app.get("/product/:productId", function(req, res){
+    var pid = parseInt(req.params.userid, 10);
+    //res.send   manipulate string to get file with name as productID or something and use a static file server
+    });
+app.listen(3000);
```

Comment:
Should the user run this code or is it just an example. I can't get an output from it.

Reply:
@louisefindlay23 I didn't write any res.send methods so you cannot see anything.

Reply:
>@rohanreddych, I receive the console message but get this error afterwards: TypeError: path must be absolute or specify root to res.sendFile

this is an example code you cannot use this "as is", it is supposed to give a basic idea

Reply:
I can write the full executable code in a "gist" and share it is ok.

Reply:
Yes, that would be great @rohanreddych.